qemu/tests/virt_firmware_basic_test.py

```diff
@@ -35,20 +39,36 @@ def run(test, params, env):
     :param env: Dictionary with test environment.
     """
 
-    virt_firmware_dirname = params["virt_firmware_dirname"]
-    virt_firmware_repo_addr = params["virt_firmware_test_repo"]
+    query_cmd = params["cmd_queried_test_package"]
+    status = process.getstatusoutput(query_cmd,
+                                     ignore_status=True,
+                                     shell=True)[0]
+    if status:
+        test.log.info("Package 'python3-virt-firmware-tests' "
+                      "has not been installed. "
+                      "Run the test with virt firmware repo.")
+        virt_firmware_dirname = params["virt_firmware_repo_dst_dir"]
+        virt_firmware_repo_addr = params["virt_firmware_repo_addr"]
+        test_file_black_list = params["test_file_black_list"].split()
+        if os.path.exists(virt_firmware_dirname):
+            shutil.rmtree(virt_firmware_dirname, ignore_errors=True)
+        try:
+            git.get_repo(uri=virt_firmware_repo_addr,
+                         destination_dir=virt_firmware_dirname)
+        except Exception as e:
+            test.error("Failed to clone the virt-firmware repo,"
+                       "the error message is '%s'." % six.text_type(e))
+    else:
+        test.log.info("Run the test with package "
+                      "'python3-virt-firmware-tests'.")
+        virt_firmware_dirname = params["virt_firmware_test_package_dir"]
+        test_file_black_list = []
```

Comment:
IMO, it is not needed.  As `test-pe.sh` is not in the test_dir.

Reply:
I think it's needed. For python3-virt-firmware-tests, if test-pe.sh in the tests directory, we need to throw an error when running it. (line 70 - line 79)

Reply:
Okay, if this is also a test scenario, ack.